### PR TITLE
Add late deadline feature 

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 
Added support for late deadline for dev portfolios. A late deadline is optional when creating a new dev portfolio.

### Notion/Figma Link 

[Notion link](https://www.notion.so/cornelldti/Late-Submissions-for-Dev-Portfolios-30da8b9dbefb4f6298906f4023b6f38c)

### Test Plan 
- Updated backend unit testing 
- Test validation for creating a new portfolio (i.e. late deadline is after the normal deadline, etc.) 
- Test that late submissions are marked as late (and that submission before the deadline isn't marked late)
- Test that late submissions have a `*` by the submission status 
- Test that a dev portfolio should behave normally if the late deadline wasn't set 
- Test that you can't submit a portfolio past the late deadline 
- Test that late deadline is used (only if it was set) as the "end date" for submission validation 

### Notes 
- ~~CSV exporting isn't implemented in this PR. This is currently blocked by #345.~~
- [This issue](https://www.notion.so/cornelldti/DP-Admin-Add-validation-for-creating-new-DP-assignment-66f97764abbc4bed89900ce5b8416e3b) should be addressed in a separate PR still

### CODE REVIEW

File: common-types/index.d.ts [hunk 1]
```diff
@@ -144,10 +144,12 @@ interface DevPortfolio {
...
   submissions: DevPortfolioSubmission[];
   readonly uuid: string;
+  lateDeadline?: number;
 }
 
 interface DevPortfolioSubmission {
   member: IdolMember;
   openedPRs: PullRequestSubmission[];
   reviewedPRs: PullRequestSubmission[];
+  isLate?: boolean;
```
Comment (JacksonStaniec) [inline common-types/index.d.ts:154]: Will this run into the same issue as #347 where the db can't handle fields that are `undefined`?

**Review by JacksonStaniec — COMMENTED** [review]
I think that the submission form validation still needs to be modified to allow submissions after the deadline and to actually populate `isLate` in order to see whether `*` is next to each late submission.

**Comment by JacksonStaniec** [comment]
Also there is an error that prevents creating an assignment without a late deadline. 

> TypeError: Cannot read properties of undefined (reading 'uuid')

We could also not make it "optional," and just default set the late deadline to be the same as the regular deadline.

**Comment by henry-li-06** [comment]
> I think that the submission form validation still needs to be modified to allow submissions after the deadline and to actually populate `isLate` in order to see whether `*` is next to each late submission.

Regarding populating the `isLate` field, it's done on the backend [here](https://github.com/cornell-dti/idol/blob/dp/late-submissions/backend/src/API/devPortfolioAPI.ts#L90-L93).

File: common-types/index.d.ts (hunk 1 above)
Comment (henry-li-06) [reply common-types/index.d.ts:154]: On the backend, `isLate` is always set [here](https://github.com/cornell-dti/idol/blob/dp/late-submissions/backend/src/API/devPortfolioAPI.ts#L90-L93) so we would never try to set `undefined` into the db.

**Comment by JacksonStaniec** [comment]
Since we put the "late" asterisk next to the status, and the status doesn't show for users currently, there is no feedback for the user on whether their assignment is late. Maybe we could put it next to their name in the user view?
image

**Comment by henry-li-06** [comment]
> Since we put the "late" asterisk next to the status, and the status doesn't show for users currently, there is no feedback for the user on whether their assignment is late. Maybe we could put it next to their name in the user view?
> 
> image
> 
> 

Are planning on showing submission status for portfolio 2?

**Comment by JacksonStaniec** [comment]
> > Since we put the "late" asterisk next to the status, and the status doesn't show for users currently, there is no feedback for the user on whether their assignment is late. Maybe we could put it next to their name in the user view?
> > image
> 
> Are we planning on showing submission status for portfolio 2?

Yes, if the admin override feature is working as expected. In the PR where we switch the view of the users, we can change the asterisk to show like it does for the admin view.

**Review by JacksonStaniec — APPROVED** [review]
Great work!

**Pull request merged**